lib/AST/Type.cpp

```diff
@@ -4910,6 +4910,134 @@ TypeBase::getAutoDiffTangentSpace(LookupConformanceFn lookupConformance) {
   return cache(None);
 }
 
+// Creates an `AnyFunctionType` from the given parameters, result type,
+// generic signature, and `ExtInfo`.
+static AnyFunctionType *
+makeFunctionType(ArrayRef<AnyFunctionType::Param> parameters, Type resultType,
```

Comment:
This is a general utility for AST function types. Consider adding it to `AnyFunctionType`, or just making it a closure inside `getAutoDiffDerivativeFunctionType`.